site/source/pages/POCSearchCodeAPE.tsx

```diff
@@ -0,0 +1,231 @@
+import UFuzzy from '@leeoniya/ufuzzy'
+import { useEffect, useRef, useState, Fragment } from 'react'
+
+import { TextField } from '@/design-system'
+import { CardContainer } from '@/design-system/card/Card'
+import { H3, H4 } from '@/design-system/typography/heading'
+import { Ul } from '@/design-system/typography/list'
+import { Body } from '@/design-system/typography/paragraphs'
+
+import testData from './test.json'
+
+interface CommonData<Type extends string> {
+	type: Type
+	code: string
+	title: string
+	data: string[]
+	contenuCentral: string[]
+	contenuAnnexe: string[]
+	contenuExclu: string[]
+	parent?: number
+}
+
+type Section = CommonData<'section'>
+type Division = CommonData<'division'>
+type Groupe = CommonData<'groupe'>
+type Classe = CommonData<'classe'>
+type SousClasse = CommonData<'sousClasse'>
+type Catégorie = CommonData<'catégorie'>
+type SousCatégorie = CommonData<'sousCatégorie'>
+
+type Data =
+	| Section
+	| Division
+	| Groupe
+	| Classe
+	| SousClasse
+	| Catégorie
+	| SousCatégorie
+
+const fuzzy = new UFuzzy({
+	intraIns: 2,
+})
+
+const filteredData = (testData as Data[]).filter(
+	({ type }) => type === 'sousClasse'
+)
+
+const { specificList, genericList, excludedList } = filteredData.reduce(
+	(prev, { contenuCentral, contenuAnnexe, contenuExclu, title, code }) => ({
+		specificList: [
+			...(prev.specificList ?? []),
+			[title, ...contenuCentral, code].join(', '),
+		],
+		genericList: [...(prev.genericList ?? []), contenuAnnexe.join(', ')],
+		excludedList: [...(prev.excludedList ?? []), contenuExclu.join(', ')],
+	}),
+	{} as {
+		genericList: string[]
+		specificList: string[]
+		excludedList: string[]
+	}
+)
+
+const latinizedSpecificList = UFuzzy.latinize(specificList)
+const latinizedGenericList = UFuzzy.latinize(genericList)
+const latinizedExcludedList = UFuzzy.latinize(excludedList)
+
+export default function POCSearchCodeAPE() {
+	const [value, setValue] = useState('')
+	// const [list, setList] = useState<[number, ...string[]][]>([])
+	const [list, setList] = useState<Data[]>([])
+
+	const lastIdxs = useRef<Record<string, UFuzzy.HaystackIdxs>>({})
+	const prevValue = useRef<string>(value)
+
+	const highlight = (str: string, range: number[]) => [
+		str.slice(0, range[0]),
+		...range
+			.map((a, i) =>
+				range[i + 1] !== undefined ? ([a, range[i + 1]] as const) : null
+			)
+			.filter(<T,>(x: T | null): x is T => x != null)
+			.map(([a, b]) => str.slice(a, b)),
```

Comment:
difficule à lire et on fait 3 itérations, un reduce doit pouvoir peut être faire le travail en une itération

Reply:
La fonction `highlight` n'est plus et ne sera pas utilisé, c'est juste un oublie de suppression